File: src/main/java/com/cfgglobal/test/config/json/JsonConfigItem.java
package com.cfgglobal.test.config.json;

import com.querydsl.core.types.Path;
import io.vavr.collection.List;
import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Accessors
class JsonConfigItem {

    Class<?> type;
    List<Path> include = List.empty();
    List<Path> exclude = List.empty();
}

File: src/main/kotlin/com/cfgglobal/test/config/json/JsonConfigItem.kt
package com.cfgglobal.test.config.json

import com.querydsl.core.types.Path

data class JsonConfigItem(
        var type: Class<*>? = null,
        var include: MutableList<Path<*>> = mutableListOf(),
        var exclude: MutableList<Path<*>> = mutableListOf()
)

